### ISSUE TITLE
Nav Unification with Polylang : posts page excludes posts in languages other than the default

### ISSUE THREAD

**Issue by rantoncuadrado** [Automattic/wp-calypso#49165]
### Steps to reproduce the behavior

You need a running Polylang installation. I imagine this would happen with other plugins making changes to posts too. My default language is SPanish, but I have French and ENglish too.
![image](https://user-images.githubusercontent.com/6404071/105454543-5a92b500-5c82-11eb-8516-8730c51260b0.png)

1. Go to the posts page https://wordpress.com/posts/remotefrog.com and you'll see just SPANISH posts
![image](https://user-images.githubusercontent.com/6404071/105454670-89a92680-5c82-11eb-8863-4336b1fb9ecd.png)
2. If you compare it with https://remotefrog.com/wp-admin/edit.php?post_status=publish&post_type=post, you'll see the difference:
![image](https://user-images.githubusercontent.com/6404071/105454760-b78e6b00-5c82-11eb-8623-21ff89007c3a.png)

3. If I go to https://remotefrog.com/wp-admin/admin.php?page=mlang (Polylang options) and change my default language to English, I'll just see the English posts in https://wordpress.com/posts/remotefrog.com
![image](https://user-images.githubusercontent.com/6404071/105454894-f3c1cb80-5c82-11eb-9669-9e3cf710a54a.png)
And this is the result:
![image](https://user-images.githubusercontent.com/6404071/105454946-0805c880-5c83-11eb-943a-603b73488d67.png)

This is a old behavior, but before the unification, Polylang users would just go to wp-admin and click on posts. Now this is not possible and you need to use the URL https://remotefrog.com/wp-admin/edit.php?post_status=publish&post_type=post to see all posts. 

#### What I expected to happen
We should display all the posts in the view

#### What actually happened
Just the main language posts are shown

### Context
I understand this is not properly said 'our' bug, but I'd say it's a serious issues because:

- If it affects Polylang it could also affect other plugins
- It creates a problematic experience for a kind of user that blogs usually and in several languages. (Potentially also to those using polylang with woocommerce)

#### Does this happen on simple or atomic sites or both?
Polylang plugin can just be installed on atomic sites.

**Comment by cpapazoglou** [comment]
This one will probably be resolved by https://github.com/Automattic/wp-calypso/issues/47043.

TL;DR; Customers that are used or need to use wp-admin screens may set this behavior as default.

**Comment by rantoncuadrado** [comment]
@cpapazoglou I don't think this solves it (It gives an alternative: using wp-admin, but the fact that all these posts are not shown in calypso would continue to happen)

**Comment by cpapazoglou** [comment]
The posts link would eventually link to `wp-admin` isn't that enough?

eg Clicking "All Posts" ->  `https://remotefrog.com/wp-admin/edit.php?post_type=post`

**Comment by rantoncuadrado** [comment]
I meant Polylang users could not be able to use the new dashboard in all its shininess. But, yes, I understand what you mean.
